components/brave_wallet/browser/solana_tx_manager.cc

```diff
@@ -35,6 +36,14 @@ namespace brave_wallet {
 // https://docs.rs/solana-program/1.18.10/src/solana_program/clock.rs.html#129-131
 constexpr int kValidBlockHeightThreshold = 150;
 
+// Transactions submitted after 30 minutes are safe to be updated to dropped
+// since usual transactions are only valid for about 2 minutes.
+// Most transactions are updated to dropped after blockhash is expired (under 2
+// minutes). This is severd as a fallbacks to cleanup transactions which are
+// stuck in submitted status somehow, for example, old transactions without
+// last valid block height information.
```

Comment:
What's the expected behavior for this if the RPC is down and not relaying transaction for us?

AFAICT this would still work, but that seemed like the one weird state that might not work well with a long delay depending on how it behaves with the block height.

Reply:
The current UX is unchanged, the transaction will be updated to dropped status and you're able to retry manually via finding the entry in the transaction history.

If that does happen, currently the transactions would be stuck at submitted status because we cannot get blockhash and hence block_tracker won't trigger OnLatestBlockhashUpdated to update pending transactions. Once we're able to get blockhash from RPC, those transactions would now be updated to dropped.
Though it's probably very rare and also most things won't function well (seems to be stuck to the user) in this case.